src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = {};
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const dateArray = date.split(fromSeparator);
+
+  fromFormat.slice(0, -1).forEach((part, index) => {
+    dateParts[part] = dateArray[index];
+  });
+
+  // Перетворення року, якщо потрібно
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+    delete dateParts['YYYY'];
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = parseInt(dateParts['YY'], 10);
+
+    dateParts['YYYY'] =
+      year < 30 ? `20${dateParts['YY']}` : `19${dateParts['YY']}`;
+    delete dateParts['YY'];
+  }
+
+  // Збірка нової дати з новим роздільником
+  const newSeparator = toFormat[toFormat.length - 1];
```

Comment:
Similar to the `fromFormat`, the assumption that the last element of `toFormat` is the separator should be verified. If the separator is not always the last element, this logic might not work as intended.